src/main/java/com/aws/greengrass/builtin/services/pubsub/SubscribeRequest.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.builtin.services.pubsub;
+
+import lombok.Builder;
+import lombok.Value;
+import software.amazon.awssdk.aws.greengrass.model.ReceiveMode;
+
+@Builder
+@Value
+public class SubscribeRequest {
+    String topic;
+    Object callback;
+    String serviceName;
+    ReceiveMode receiveMode;
```

Comment:
do you need to exclude receiveMode from `equals` and `hashcode`? Ensure that unsubscribe actually works when using the old API (not providing a SubscribeRequest object, but instead just proving a callback, topic, etc)

Reply:
Currently it is implemented that the old API unsubscribe() has the same steps as the old subscribe() so their SubscribeRequest will be the same: it is treated as having no receive mode and it evaluates the mode based on if it's using wildcard, and generates a `SubscriptionCallback` object that also includes the ReceiveMode. The trie will remove that subscription given the topic and `SubscriptionCallback` so ReceiveMode will also be compared when removing. I've also thought about if we should just ignore the ReceiveMode when unsubscribe but I thought we would want to check it..
`GIVEN_subscribed_consumer_no_receive_mode_WHEN_publish_binary_message_THEN_publishes_And_THEN_unsubscribes` and `GIVEN_subscribed_consumer_to_wildcard_and_no_receive_mode_WHEN_publish_to_subtopic_THEN_not_publishes_And_THEN_unsubscribes` tests the old subscribe() and unsubscribe().